apps/src/codebridge/FileBrowser/InnerFileBrowser.tsx

```diff
@@ -120,76 +102,4 @@ const InnerFileBrowser = React.memo(
   }
 );
 
-export const FileBrowser = React.memo(() => {
-  const {source, setFileType} = useCodebridgeContext();
-  const isReadOnly = useAppSelector(isReadOnlyWorkspace);
-  const appName = useAppSelector(state => state.lab.levelProperties?.appName);
-
-  const [dragData, setDragData] = useState<DragDataType | undefined>(undefined);
-  const [dropData, setDropData] = useState<DropDataType | undefined>(undefined);
-
-  const dndMonitor = useMemo(
-    () => ({
-      onDragStart: (e: DragStartEvent) =>
-        setDragData(e.active.data.current as DragDataType),
-      onDragOver: (e: DragOverEvent) =>
-        setDropData(e.over?.data.current as DropDataType),
-      onDragEnd: (e: DragEndEvent) => {
-        setDragData(undefined);
-        setDropData(undefined);
-      },
-    }),
-    [setDragData, setDropData]
-  );
-
-  const handleDragEnd = useHandleDragEnd();
-
-  const sensors = useSensors(
-    useSensor(PointerSensor, {
-      activationConstraint: {
-        distance: 2,
-      },
-    })
-  );
-
-  return (
-    <PanelContainer
-      id="file-browser"
-      headerContent={codebridgeI18n.filesHeader()}
-      headerClassName={moduleStyles.fileBrowserHeader}
-      className={moduleStyles['file-browser']}
-      rightHeaderContent={!isReadOnly && <FileBrowserHeaderPopUpButton />}
-    >
-      <div className={moduleStyles.fileBrowserContents}>
-        <DndContext onDragEnd={handleDragEnd} sensors={sensors}>
-          <DndDataContextProvider
-            value={{dragData, dropData}}
-            dndMonitor={dndMonitor}
-          >
-            <Droppable
-              data={{id: DEFAULT_FOLDER_ID}}
-              className={classNames(
-                moduleStyles.droppableArea,
-                moduleStyles.expandedDroppableArea,
-                {
-                  [moduleStyles.acceptingDrop]:
-                    DEFAULT_FOLDER_ID === dropData?.id,
-                }
-              )}
-            >
-              <ul id="uitest-files-list">
-                <InnerFileBrowser
-                  parentId={DEFAULT_FOLDER_ID}
-                  folders={source.folders}
-                  files={source.files}
-                  setFileType={setFileType}
-                  appName={appName}
-                />
-              </ul>
-            </Droppable>
-          </DndDataContextProvider>
-        </DndContext>
-      </div>
-    </PanelContainer>
-  );
-});
+export default InnerFileBrowser;
```

Comment:
Could we add a comment explaining what `InnerFileBrowser` is wrt to `FileBrowser` at top of file?

Reply:
Also, this is a really interesting component - at first I was confused by how the component references itself. I haven't seen this recursive type definition before.
Maybe add a short comment about how it is recursive for handling nested folders?

Reply:
I'll add a comment!